### PR TITLE
aws-lambda: Add types for 'Step Functions Execution Status Change'

### PR DESCRIPTION
**Pull request by photz**
Re-opening an earlier PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64688
This PR addresses the change requested by @jamesorlakin.

Source: https://docs.aws.amazon.com/step-functions/latest/dg/cw-events.html

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.aws.amazon.com/step-functions/latest/dg/cw-events.html
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by photz** [comment]
@darbio @brew0722

File: types/aws-lambda/index.d.ts [hunk 1]
```diff
@@ -92,5 +92,6 @@ export * from './trigger/self-managed-kafka';
 export * from './trigger/secretsmanager';
 export * from './trigger/s3-event-notification';
 export * from './trigger/amplify-resolver';
+export * from "./trigger/step-functions";
```
Comment (darbio) [inline types/aws-lambda/index.d.ts:95]: Please can you change these to single quotes as per the other exports.

**Comment by darbio** [comment]
@photz looks good. Can you just address the quotes and I will approve then we can merge this in.

**Review by darbio — CHANGES_REQUESTED** [review]
Please address quotes as per comment.

**Comment by simonbuchan** [comment]
Hmm, well there's also the eslint error, which is not related to the quotes (for which I would also accept running the formatter, which would change them all to double quotes, iirc)

**Pull request closed without merging**